blog/compileToJson.ts

```diff
@@ -0,0 +1,57 @@
+import path from "path";
+import fs from "fs";
+import { parseMdx } from "./parser";
+import { COMPONENT_DIR, POST_DIR } from "./paths";
+
+function readComponents() {
+  if (!fs.existsSync(COMPONENT_DIR)) {
+    return undefined;
+  }
+  return fs
+    .readdirSync(COMPONENT_DIR)
+    .filter(isTypescriptFilename)
+    .map((filename) => path.join(COMPONENT_DIR, filename))
+    .reduce((componentFiles, file) => {
+      const relativePath = path.relative(POST_DIR, file);
+      const content = fs.readFileSync(file).toString("utf-8");
+      return { ...componentFiles, [relativePath]: content };
+    }, {} as Record<string, string>);
```

Comment:
Chaining these methods together has you looping over the data 3 times. My advice would be to only use `reduce` and run the `isTypescriptFilename` function, join the paths, and return the `componentFiles` object all in the reducer.

Reply:
To your first question, yes there is support for mdx files out of the box, but even on the Remix documentation, they mention that it's rudimentary and they recommend using `mdx-bundler`. This is an attempt to follow up on @colbywhite's discussion of taking the best stab at a file based CMS that follows up on what he's made, which was also inspired by what Kent C. Dodds created for his site. The goal here is to make the start of what could be a more useful lib for the whole Remix community. This solution allows us to have a dynamic level of content without manually importing every blog post, which is what the base level of Remix offers for `.mdx`.

Reply:
As to your code comment, I'll address it tomorrow. Thanks for catching this one

Reply:
Interesting - Thanks for the explanation @matt-hernandez!

Reply:
@sethdavis512 I lied. I changed this today 😆. Let me know if it's what you suggested.

Reply:
I also thought of something else. That callout on the Remix documentation about `.mdx` files could be easy to miss. I'm going to add a section to my blog post to make sure I cover that.

That said, this PR is taking the concept of a local file-based CMS in Remix to it's natural conclusion, and I can already see some pretty big gotchas. Even if this gets adopted as an acceptable solution, I want to be quick to put some guardrails in place, and possibly even drop the idea of serving content locally and start moving to hosting blog content, most likely on Github and fetching via API.